### PR TITLE
feat: add custom grafana DockerFile

### PR DESCRIPTION
**Pull request by balajisa09**

### CODE REVIEW

File: datasources/datasources.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: 1
+datasources:
+  - name: Prometheus
+    type: prometheus
+    url: http://kps-prometheus.glueops-core-kube-prometheus-stack:9090/
```
Comment (venkatamutyala) [inline datasources/datasources.yaml]: We shouldn't need this if the local metrics are also going to cortex, right?

File: test/grafana-deployment.yaml
Comment (venkatamutyala) [inline test/grafana-deployment.yaml:1]: Do we need these test deployments/files? Or could we have an example that provides YAML that works with what we do in deployment configurations?

File: grafana.ini
Comment (venkatamutyala) [inline grafana.ini:1]: What's the default username/password? Can we do anything trivial that uses google SSO or something that just checks the domain/email of the user logging in?

File: datasources/datasources.yaml (hunk 1 above)
Comment (balajisa09) [reply datasources/datasources.yaml]: That's right. Just added this for testing and can be removed.

**Review by venkatamutyala — CHANGES_REQUESTED** [review]
See inline comments

File: test/grafana-deployment.yaml
Comment (balajisa09) [reply test/grafana-deployment.yaml:1]: yeah the test manifests can be rewritten to adopt deployment-configuration style of manifests

File: grafana.ini
Comment (balajisa09) [reply grafana.ini:1]: it is admin/admin. Yeah can look into auth options.

File: datasources/datasources.yaml (hunk 1 above)
Comment (venkatamutyala) [reply datasources/datasources.yaml]: Perfect. Let's pull it out.

File: test/grafana-deployment.yaml
Comment (venkatamutyala) [reply test/grafana-deployment.yaml:1]: If the current manifests are helpful for you in the future please feel free to leave them. I think when we deploy or when I test as well i'll just use the deployment configurations type of manifests

**Review by venkatamutyala — COMMENTED** [review]
added comments

File: test/grafana-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  template:
+    metadata:
+      labels:
+        app: grafana
+    spec:
+      imagePullSecrets:
+      - name: ghcr-secret
+      containers:
+      - name: grafana
+        image: ghcr.io/balajisa09/grafana-custom:v1.0.4
```
Comment (venkatamutyala) [inline test/grafana-deployment.yaml]: could you point this to the glueops org/image?

File: datasources/datasources.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+apiVersion: 1
+datasources:
+  - name: Cortex
+    type: prometheus
+    url: http://cortex-metrics-test.nonprod.svc.cluster.local:9009/prometheus
```
Comment (venkatamutyala) [inline datasources/datasources.yaml]: can we drop `test` and just call it cortex-metrics?

File: README.md [hunk 4]
```diff
@@ -1 +1 @@
-# template
+# Custom Grafana Image
```
Comment (venkatamutyala) [inline README.md]: Cortex Grafana Image?

**Review by venkatamutyala — CHANGES_REQUESTED** [review]

**Review by venkatamutyala — DISMISSED** [review]

**Review by venkatamutyala — APPROVED** [review]

**Pull request merged**